### PR TITLE
docs(decision-tree-widget): implement Operator Decision Tree widget

### PR DESCRIPTION
**Pull request by staltz**
Implements the Operator Decision Tree widget, and add to both doc/index.md and doc/operators.md
Notice specially `doc/decision-tree-widget/tree.yml` which is the decision tree data encoded in YAML. Probably the decision tree YAML file still needs review and improvement, but we could update it separately from this PR.

To see this page live, open http://rxjs5-esdoc-decision-tree.surge.sh/

choose

### CODE REVIEW

**Comment by blesh** [comment]
That is super cool.

**Comment by kwonoj** [comment]
Love this too. :cupid:

**Comment by todoubaba** [comment]
It seems the operator's url is not correct, such as `mapTo` links to `http://rxjs5-esdoc-decision-tree.surge.sh/class/es6/Observable.js~Observable.html`.

**Comment by staltz** [comment]
@todoubaba The links are correct, but they refer to a future version of the Observable.html page, we're depending on PR #1376 merged first.

**Comment by todoubaba** [comment]
Yeah, looking forward that :+1:

**Comment by kwonoj** [comment]
:heart: Change looks good to me.

**Comment by staltz** [comment]
I need to rebase, apparently

**Comment by staltz** [comment]
Rebased, but it will conflict with #1385 when that is merged. (and will need another rebase)

**Comment by staltz** [comment]
Rebossed.

File: doc/decision-tree-widget/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "yaml2json": "rm -f src/tree.json && node tools/yaml2json tree.yml ./src/tree.json",
+    "prebrowserify": "mkdir -p dist",
+    "browserify": "browserify src/main.js -t babelify --outfile dist/decision-tree-widget.js",
+    "build": "npm install && npm run yaml2json && npm run browserify",
+    "postbuild": "rm -f src/tree.json"
+  },
+  "dependencies": {
+    "@motorcycle/core": "^1.1.0",
+    "@motorcycle/dom": "^1.2.1",
+    "most": "^0.18.1",
```
Comment (blesh) [inline doc/decision-tree-widget/package.json]: LOL... what?! :-1:

Comment (blesh) [inline doc/decision-tree-widget/package.json]: You're just asking for @briancavalier to come tease me aren't you?

**Comment by blesh** [comment]
So, you know what I'm going to say. This is really cool. We should probably implement it with _RxJS 5_ and annotate the code. :P 

LOLOLOL... Too funny, @staltz

**Comment by staltz** [comment]
So, in reality it's not trolling and it's absolutely serious stuff. Cycle.js doesn't currently work with RxJS v5, we're working on getting that support up. In the meanwhile, RxJS v4 is quite huge in size to include to this page, specially because we already have full RxJS v5 loaded in the page. So I went for something small in size, and I chose a version of Cycle.js called Motorcycle.js, focused on performance.

If you'd prefer for PR purposes that I just use RxJS v4 even if it's larger, fine by me. Or, I can place the whole decision tree widget elsewhere, make it generic for any purposes, then just import it here. So, your call.

**Comment by blesh** [comment]
> If you'd prefer for PR purposes that I just use RxJS v4 even if it's larger, fine by me. 

Honestly, that's probably better. But ideally, we'd have it in RxJS 5 and it would be annotated.

Could write it in Angular 2 ;) haha.

**Comment by staltz** [comment]
Could write it in anything. I just wanted to get it done.

> But ideally, we'd have it in RxJS 5 and it would be annotated.

Will do when the support is out. I'll update this PR with Rx4 to make y'all happy.

File: doc/decision-tree-widget/package.json (hunk 1 above)
Comment (briancavalier) [inline doc/decision-tree-widget/package.json]: lgtm :+1: :ship:

Comment (Cmdv) [inline doc/decision-tree-widget/package.json]: This is so good, it's nearly up there with when that gentleman asked @blesh on his article, if he'd used RxJS!!
#acidental-trolling  :trollface:

**Comment by xgrommx** [comment]
@staltz Now you can use only what you want of Rx instead of whole library https://github.com/Reactive-Extensions/RxJS/tree/master/src/modular

**Comment by staltz** [comment]
I updated the PR to use Rx v4, and uglifyjs.

**Comment by staltz** [comment]
Rebased

**Comment by kwonoj** [comment]
Looks good to me (but another rebase... :sweat: )

**Comment by staltz** [comment]
Rebombed. :)

**Comment by staltz** [comment]
@blesh are you serious about this PR having to use RxJS v5 for dogfooding purposes? Just want to know what are your intentions because I could honestly convert this to a pseudo-Cycle.js app, since it's so small I can give up a bit on the actual framework and just apply the same architectural ideas.

**Comment by staltz** [comment]
Ok, I just did what I said in my previous comment, so now it should be fully *Blesh-Compliant Enterprise Software*.

**Comment by staltz** [comment]
Rebased :white_check_mark:

**Comment by kwonoj** [comment]
I think change's ok to be checked in, will check this in around tomorrow unless other suggestion's around.

**Comment by kwonoj** [comment]
Merged with ffe71df, :tada: @staltz

**Pull request closed without merging**